llvm/lib/Support/raw_socket_stream.cpp

```diff
@@ -41,6 +41,8 @@
 
 using namespace llvm;
 
+#if defined(_WIN32) || !defined(__wasi__)
```

Comment:
This conditional is weird; there's no case where `__wasi__` is defined at the same time as `_WIN32`, so you just mean `#if !defined(__wasi__)`.

Also you've removed the entire type's implementation, but not the declaration in the header, which is broken.

Reply:
`raw_socket_stream` isn't actually used anywhere except for tests, so this doesn't break anything. Should it perhaps be removed instead?